TestHeartbeatSingleGroup is flaky

The following test appears to have failed:

[#899](https://circleci.com/gh/cockroachdb/cockroach/899):

```
I0325 17:47:13.158756      81 multiraft.go:449] node 1: group 1 got message 9->1 MsgHeartbeatResp Term:6 Log:0/0
I0325 17:47:13.158795      81 multiraft.go:449] node 1: group 1 got message 10->1 MsgHeartbeatResp Term:6 Log:0/0
I0325 17:47:13.158894      81 multiraft.go:449] node 1: group 1 got message 3->1 MsgHeartbeatResp Term:6 Log:0/0
I0325 17:47:13.158932      81 multiraft.go:449] node 1: group 1 got message 4->1 MsgHeartbeatResp Term:6 Log:0/0
I0325 17:47:13.158993      81 multiraft.go:449] node 1: group 1 got message 5->1 MsgHeartbeatResp Term:6 Log:0/0
--- FAIL: TestHeartbeatSingleGroup (0.84s)
	heartbeat_test.go:235: actual and expected heartbeat counts differ for 10 nodes, 0 leader ticks:
		{ 1: [reqOut=18 reqIn=0 respOut=0 respIn=0], 2: [reqOut=0 reqIn=2 respOut=0 respIn=0], 3: [reqOut=0 reqIn=2 respOut=0 respIn=0], 4: [reqOut=0 reqIn=2 respOut=0 respIn=0], 5: [reqOut=0 reqIn=2 respOut=0 respIn=0], 6: [reqOut=0 reqIn=2 respOut=0 respIn=0], 7: [reqOut=0 reqIn=2 respOut=0 respIn=0], 8: [reqOut=0 reqIn=2 respOut=0 respIn=0], 9: [reqOut=0 reqIn=2 respOut=0 respIn=0], 10: [reqOut=0 reqIn=2 respOut=0 respIn=0], }
		{ 1: [reqOut=18 reqIn=1 respOut=0 respIn=0], 2: [reqOut=9 reqIn=2 respOut=0 respIn=0], 3: [reqOut=0 reqIn=3 respOut=0 respIn=0], 4: [reqOut=0 reqIn=3 respOut=0 respIn=0], 5: [reqOut=0 reqIn=3 respOut=0 respIn=0], 6: [reqOut=0 reqIn=3 respOut=0 respIn=0], 7: [reqOut=0 reqIn=3 respOut=0 respIn=0], 8: [reqOut=0 reqIn=3 respOut=0 respIn=0], 9: [reqOut=0 reqIn=3 respOut=0 respIn=0], 10: [reqOut=0 reqIn=3 respOut=0 respIn=0], }
=== RUN TestHeartbeatMultipleGroupsJointLeader
I0325 17:47:13.159571      81 multiraft.go:408] node 1 starting
I0325 17:47:13.159628      81 multiraft.go:408] node 2 starting
I0325 17:47:13.159681      81 multiraft.go:408] node 3 starting
I0325 17:47:13.159730      81 multiraft.go:408] node 4 starting
I0325 17:47:13.159779      81 multiraft.go:408] node 5 starting
I0325 17:47:13.159834      81 multiraft.go:408] node 6 starting
--
I0325 17:47:13.275984      81 multiraft.go:637] Committed Entry[0]: 6/16 EntryConfChange "\b\x00\x10\x00\x18\x04\"\x11\x00zSIEBErBLLOCwUNA"
I0325 17:47:13.276084      81 multiraft.go:643] Outgoing Message[0]: 1->3 MsgSnap Term:6 Log:0/0 Snapshot:{[] {{[1] []} 10 5 []} []}
I0325 17:47:13.276121      81 multiraft.go:643] Outgoing Message[1]: 1->2 MsgApp Term:6 Log:6/16 Commit:16
I0325 17:47:13.276309      81 raft.go:484] raft: 1 snapshot succeeded, resumed sending replication messages to 3 [next = 11, match = 0, state = ProgressStateProbe, waiting = false, pendingSnapshot = 0]
--- PASS: TestMembershipChange (0.03s)
FAIL
FAIL	github.com/cockroachdb/cockroach/multiraft	0.983s
=== RUN TestMemoryStorage
--- PASS: TestMemoryStorage (0.00s)
PASS
ok  	github.com/cockroachdb/cockroach/multiraft/storagetest	0.006s
=== RUN TestClientCmdIDIsEmpty
--- PASS: TestClientCmdIDIsEmpty (0.00s)
=== RUN TestResponseHeaderSetGoError
--- PASS: TestResponseHeaderSetGoError (0.00s)
=== RUN TestResponseHeaderNilError
--- PASS: TestResponseHeaderNilError (0.00s)
I0325 17:47:13.158756      81 multiraft.go:449] node 1: group 1 got message 9->1 MsgHeartbeatResp Term:6 Log:0/0
I0325 17:47:13.158795      81 multiraft.go:449] node 1: group 1 got message 10->1 MsgHeartbeatResp Term:6 Log:0/0
I0325 17:47:13.158894      81 multiraft.go:449] node 1: group 1 got message 3->1 MsgHeartbeatResp Term:6 Log:0/0
I0325 17:47:13.158932      81 multiraft.go:449] node 1: group 1 got message 4->1 MsgHeartbeatResp Term:6 Log:0/0
I0325 17:47:13.158993      81 multiraft.go:449] node 1: group 1 got message 5->1 MsgHeartbeatResp Term:6 Log:0/0
--- FAIL: TestHeartbeatSingleGroup (0.84s)
	heartbeat_test.go:235: actual and expected heartbeat counts differ for 10 nodes, 0 leader ticks:
		{ 1: [reqOut=18 reqIn=0 respOut=0 respIn=0], 2: [reqOut=0 reqIn=2 respOut=0 respIn=0], 3: [reqOut=0 reqIn=2 respOut=0 respIn=0], 4: [reqOut=0 reqIn=2 respOut=0 respIn=0], 5: [reqOut=0 reqIn=2 respOut=0 respIn=0], 6: [reqOut=0 reqIn=2 respOut=0 respIn=0], 7: [reqOut=0 reqIn=2 respOut=0 respIn=0], 8: [reqOut=0 reqIn=2 respOut=0 respIn=0], 9: [reqOut=0 reqIn=2 respOut=0 respIn=0], 10: [reqOut=0 reqIn=2 respOut=0 respIn=0], }
		{ 1: [reqOut=18 reqIn=1 respOut=0 respIn=0], 2: [reqOut=9 reqIn=2 respOut=0 respIn=0], 3: [reqOut=0 reqIn=3 respOut=0 respIn=0], 4: [reqOut=0 reqIn=3 respOut=0 respIn=0], 5: [reqOut=0 reqIn=3 respOut=0 respIn=0], 6: [reqOut=0 reqIn=3 respOut=0 respIn=0], 7: [reqOut=0 reqIn=3 respOut=0 respIn=0], 8: [reqOut=0 reqIn=3 respOut=0 respIn=0], 9: [reqOut=0 reqIn=3 respOut=0 respIn=0], 10: [reqOut=0 reqIn=3 respOut=0 respIn=0], }
=== RUN TestHeartbeatMultipleGroupsJointLeader
I0325 17:47:13.159571      81 multiraft.go:408] node 1 starting
I0325 17:47:13.159628      81 multiraft.go:408] node 2 starting
I0325 17:47:13.159681      81 multiraft.go:408] node 3 starting
I0325 17:47:13.159730      81 multiraft.go:408] node 4 starting
I0325 17:47:13.159779      81 multiraft.go:408] node 5 starting
I0325 17:47:13.159834      81 multiraft.go:408] node 6 starting
--
I0325 17:47:13.275984      81 multiraft.go:637] Committed Entry[0]: 6/16 EntryConfChange "\b\x00\x10\x00\x18\x04\"\x11\x00zSIEBErBLLOCwUNA"
I0325 17:47:13.276084      81 multiraft.go:643] Outgoing Message[0]: 1->3 MsgSnap Term:6 Log:0/0 Snapshot:{[] {{[1] []} 10 5 []} []}
I0325 17:47:13.276121      81 multiraft.go:643] Outgoing Message[1]: 1->2 MsgApp Term:6 Log:6/16 Commit:16
I0325 17:47:13.276309      81 raft.go:484] raft: 1 snapshot succeeded, resumed sending replication messages to 3 [next = 11, match = 0, state = ProgressStateProbe, waiting = false, pendingSnapshot = 0]
--- PASS: TestMembershipChange (0.03s)
FAIL
FAIL	github.com/cockroachdb/cockroach/multiraft	0.983s
=== RUN TestMemoryStorage
--- PASS: TestMemoryStorage (0.00s)
PASS
ok  	github.com/cockroachdb/cockroach/multiraft/storagetest	0.006s
=== RUN TestClientCmdIDIsEmpty
--- PASS: TestClientCmdIDIsEmpty (0.00s)
=== RUN TestResponseHeaderSetGoError
--- PASS: TestResponseHeaderSetGoError (0.00s)
=== RUN TestResponseHeaderNilError
--- PASS: TestResponseHeaderNilError (0.00s)

```
Please assign, take a look and update the issue accordingly.

--- a/multiraft/heartbeat_test.go
+++ b/multiraft/heartbeat_test.go
@@ -37,8 +37,8 @@ func processEventsUntil(ch <-chan *interceptMessage, stopper *util.Stopper,
 	if stopper != nil {
 		defer stopper.SetStopped()
 	}
-	t := time.After(time.Second)
 	for {
+		t := time.After(time.Second)
 		select {
 		case e, ok := <-ch:
 			if !ok {

